logstage/logstage-core/src/main/scala/izumi/logstage/api/strict/IzStrictLogger.scala

```diff
@@ -3,22 +3,25 @@ package izumi.logstage.api.strict
 import izumi.logstage.api.Log.CustomContext
 import izumi.logstage.api.logger.{AbstractMacroStrictLogger, LogRouter, RoutingLogger}
 import izumi.logstage.api.rendering.StrictEncoded
-import izumi.logstage.api.{IzLogger, IzLoggerConvenienceApi, Log}
+import izumi.logstage.api.{IzLoggerConvenienceApi, IzRawLogger, Log}
 
 class IzStrictLogger(
   override val router: LogRouter,
   override val customContext: Log.CustomContext,
 ) extends RoutingLogger
   with AbstractMacroStrictLogger {
 
-  override def withCustomContext(context: CustomContext): IzLogger = new IzLogger(router, customContext + context)
-  final def withCustomContext(context: (String, StrictEncoded)*): IzLogger = withCustomContextMap(context.toMap)
-  final def withCustomContextMap(context: Map[String, StrictEncoded]): IzLogger = withCustomContext(CustomContext.fromMap(context))
+  override type Self = IzStrictLogger
 
-  final def apply(context: CustomContext): IzLogger = withCustomContext(context)
```

Comment:
Notice an issue here: a strict logger could become unstrict. That's why (and not only) I introduced the `Self` type.